edxlearndot/learndot.py

```diff
@@ -339,38 +346,124 @@ def get_enrolment_id(self, contact_id, component_id):
             headers=self.get_api_request_headers(),
             json={"status": status}
         )
+
         try:
             response.raise_for_status()
         except requests.exceptions.HTTPError as e:
             msg = "Error trying to set status of enrolment {} to {}: {}".format(enrolment_id, status, e)
             log.error(msg)
             raise LearndotAPIException(msg)
+
+        try:
+            enrolment_status_log, _created = EnrolmentStatusLog.objects.get_or_create(
+                learndot_enrolment_id=enrolment_id
+            )
+            enrolment_status_log.status = status
+            enrolment_status_log.save()
+            log.info(
+                "Recorded status of enrolment %s as %s at %s",
+                enrolment_status_log.learndot_enrolment_id,
+                enrolment_status_log.status,
+                enrolment_status_log.updated_at
+            )
+        except (IntegrityError, MultipleObjectsReturned) as e:
+            log.error("Error recording enrolment status update: %s", e)
+
+    def set_enrolment_status_to_passed(self, enrolment_id, unconditional=False):
+        """
+        Sets the status of a Learndot enrollment record to PASSED.
+        Arguments:
+            enrolment_id (int): the numeric Learndot enrollment ID
+            unconditional (bool): whether to unconditionally send the new
+                                  status to Learndot, instead of checking
+                                  EnrolmentStatusLog records to see if it
+                                  has already been sent.
+        Returns:
+            None
+        Raises:
+            LearndotAPIException: if the requested status is invalid, or
+                Requests throws anything.
+        """
+        self.set_enrolment_status(enrolment_id, EnrolmentStatus.PASSED, unconditional)
+
+    def check_if_enrolment_and_set_status_to_passed(self, contact_id, component_id, unconditional=False):
+        """
+        Sets the status of a Learndot enrollment record to PASSED if enrollment_id is found.
+        Arguments:
+            contact_id (int): the numeric Learndot enrollment ID
+            component_id (int): the numeric Learndot component ID.
+            unconditional (bool): whether to unconditionally send the new
+                                  status to Learndot, instead of checking
+                                  EnrolmentStatusLog records to see if it
+                                  has already been sent.
+        Returns:
+            None
+        Raises:
+            LearndotAPIException: if the requested status is invalid, or
+                Requests throws anything.
+        """
+        enrolment_id = self.get_enrolment_id(contact_id, component_id)
+        if not enrolment_id:
+            log.error("No enrolment found for contact %s, component %s", contact_id, component_id)
+            return
+        self.set_enrolment_status_to_passed(enrolment_id, unconditional=unconditional)
```

Comment:
Can you make this just call `set_enrollment_status` and remove the `set_enrolment_status_to_passed` method, since this is the only place it's now called?

This is just our API, so removing things is ok.